test/connectivity/modules/NetworkController/LinuxNetworkInterfaceOfflineController.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace NetworkController
+{
+    using System;
+    using System.Net.NetworkInformation;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    // TODO: This class is disabling the network interface which causes connection drop between all containers
+    // should be changed to use a different method
+    class LinuxNetworkInterfaceOfflineController : INetworkController
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LinuxNetworkInterfaceOfflineController>();
+        readonly string interfaceName;
+
+        public LinuxNetworkInterfaceOfflineController(string interfaceName)
+        {
+            this.interfaceName = interfaceName;
+        }
+
+        public string Description => "LinuxNetworkInterfaceOffline";
+
+        public Task<NetworkStatus> GetStatus(CancellationToken cs)
+        {
+            OperationalStatus status = this.GetNetworkInterfaceStatus();
+            switch (status)
+            {
+                case OperationalStatus.Up:
+                    return Task.FromResult(NetworkStatus.Default);
+                case OperationalStatus.Unknown:
```

Comment:
nit: should it be better that Unknown is default case instead?

Reply:
Unknown doesn't mean is Restricted, but all other cases somehow means that is restricted.